samtranslator/model/intrinsics.py

```diff
@@ -30,6 +30,24 @@ def make_conditional(condition, data):
     }
 
 
+def make_if_conditional(condition, true_data, false_data):
```

Comment:
This is just repeating the previous `make_conditional`- we could override that one instead like this:

`def make_conditional(condition, true_data, false_data={'Ref': 'AWS::NoValue'}):`

Keeps the same functionality and doesn't copy code.